### PR TITLE
use staticcompiler for julia instead of packagecompiler.jl

### PR DESCRIPTION
**Pull request by Moelf**

### CODE REVIEW

**Comment by Moelf** [comment]
this compiles in a few seconds and runs like C (i.e. no start-up overhead):
```
/t/s/src (master)> hyperfine ./mainjl
Benchmark 1: ./mainjl
  Time (mean ± σ):     116.6 ms ±   1.8 ms    [User: 116.2 ms, System: 0.5 ms]
  Range (min … max):   114.1 ms … 119.0 ms    25 runs
```

**Review by niklas-heer — APPROVED** [review]

**Comment by Moelf** [comment]
`ld` couldn't find `-lgcc` even though we installed `gcc` and `build-base`, according to:
- https://github.com/gliderlabs/docker-alpine/issues/24#issuecomment-710659397
`alpine-sdk` is a superset, so let's try to see if that resolves out build dependency

File: src/leibniz_compiled.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    buf = MallocString(undef, 16)
+    StaticTools.fread!(buf, f)
+    fclose(f)
+    rounds = parse(Int64, buf)
+    free(buf)
+    
+    x = 1.0
+    pi = 1.0
+
+    for i in 2:(rounds + 2)
```
Comment (mcabbott) [inline src/leibniz_compiled.jl:12]: Will it still work if this actual computation is a separate function (the same as before) called by `mainjl` which does read/write? That might be prettier / more julian.

Comment (Moelf) [reply src/leibniz_compiled.jl:12]: oh, yes, yes it would, actually we could source the original `jl` file from the `compiled.jl` file

Comment (Moelf) [reply src/leibniz_compiled.jl]: @mcabbott done

**Comment by niklas-heer** [comment]
Even after adding all of this:
```
RUN apk add --no-cache clang lld libc-dev musl-dev compiler-rt compiler-rt-static alpine-sdk gcc hdf5-dev cmake
```
It didn't work. But with Debian, it does work. Adding a working commit.

**Review by niklas-heer — APPROVED** [review]

**Comment by Moelf** [comment]
yeah something is not very robust in Static*.jl stuff, we can do this as a workaround before the fix, thanks btw!

**Comment by Moelf** [comment]
combined_results

looking sharp

**Comment by niklas-heer** [comment]
@Moelf you're very welcome. It could also be because of Alpine they use musl instead of something else the other distros use as far as I remember. I'm a bit fuzzy on the details.
Yes, looks sharp. Going to merge it.
Thank you again for your contribution. 👍

**Comment by niklas-heer** [comment]
Ah, found [an article.](https://bell-sw.com/announcements/2022/08/03/alpine-linux-small-and-powerful-distro-for-docker-images/#musl) It's musl instead of glibc, and they are using busybox which was developed for embedded systems. That's why it is such a tiny image (~3MB see [latest image](https://hub.docker.com/layers/library/alpine/latest/images/sha256-1304f174557314a7ed9eddb4eab12fed12cb0cd9809e4c28f29af86979a3c870?context=explore)), but sometimes software doesn't work because of these shortcuts, so then it's good to fall back to Debian or Ubuntu.

**Pull request merged**